Documentation/ApiOverview/Database/ConnectionPool/Index.rst

```diff
@@ -1,56 +1,57 @@
-.. include:: /Includes.rst.txt
+..  include:: /Includes.rst.txt
 
-.. _database-connection-pool:
+..  _database-connection-pool:
 
 ==============
 ConnectionPool
 ==============
 
-TYPO3's interface to execute queries via Doctrine DBAL typically starts by asking
-the `ConnectionPool` for a `QueryBuilder` or a `Connection` object, handing over the table name to be queried:
+TYPO3's interface for executing queries via Doctrine DBAL typically starts with
+a request to the :php:`ConnectionPool` for a :php:`QueryBuilder` or a
+:php:`Connection` object and passing the table name to be queried:
 
-.. code-block:: php
-   :caption: EXT:some_extension/Classes/SomeClass.php
-
-   // use TYPO3\CMS\Core\Utility\GeneralUtility;
-   // use TYPO3\CMS\Core\Database\ConnectionPool;
-   // Get a query builder for a table
-   $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_myext_comments');
-   // or
-   // Get a connection for a table
-   $connection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('tx_myext_comments');
-
-
-The `QueryBuilder` is the default workhorse object used by extension authors to express complex queries,
-while a `Connection` instance can be used as shortcut to deal with some simple query cases and little written down code.
+..  literalinclude:: _MyTableRepository.php
+    :caption: EXT:my_extension/Classes/Domain/Model/MyTableRepository.php
 
+The :php:`QueryBuilder` is the default workhorse object used by extension
+authors to express complex queries, while a :php:`Connection` instance can be
+used as a shortcut to handle some simple query cases and little written code.
```

Comment:
This sentence sounds quite strange

Reply:
Removed the last part ("and little written code").